### ISSUE TITLE
Need to handle :cflags argument for jpm/declare-executable method (/usr/bin/ld  'undefined reference' error)

### ISSUE THREAD

**Issue by kixus** [janet-lang/janet#275]
While building an executable that has a dependency on a custom Janet-C wrapper for a system shared library already installed on Linux, `/usr/bin/ld` would throw an error with failure to build the executable, stating `undefined reference` etc. 
This is observed with the latest  Janet 1.6.0 installation.  E.g. for a custom in-house `/usr/lib64/libetpan.so` wrapper, we get the following `jpm --verbose build` errors -

```bash
/usr/bin/ld: build/etpan.a(lib___etpan.static.o): in function `lib_version':
etpan.c:(.text+0x24): undefined reference to `libetpan_get_version_major'
/usr/bin/ld: etpan.c:(.text+0x2c): undefined reference to `libetpan_get_version_minor'
/usr/bin/ld: build/etpan.a(lib___etpan.static.o): in function `fetch_msg':
etpan.c:(.text+0x1d0): undefined reference to `mailimap_set_new_single'
/usr/bin/ld: etpan.c:(.text+0x1d8): undefined reference to `mailimap_fetch_type_new_fetch_att_list_empty'
/usr/bin/ld: etpan.c:(.text+0x1e2): undefined reference to `mailimap_section_new'
/usr/bin/ld: etpan.c:(.text+0x1ea): undefined reference to `mailimap_fetch_att_new_body_peek_section'
/usr/bin/ld: etpan.c:(.text+0x1f5): undefined reference to `mailimap_fetch_type_new_fetch_att_list_add'
/usr/bin/ld: etpan.c:(.text+0x208): undefined reference to `mailimap_uid_fetch'
/usr/bin/ld: etpan.c:(.text+0x2b9): undefined reference to `mailimap_fetch_list_free'
/usr/bin/ld: etpan.c:(.text+0x30b): undefined reference to `mailimap_fetch_list_free'
/usr/bin/ld: etpan.c:(.text+0x361): undefined reference to `mailimap_fetch_list_free'
/usr/bin/ld: build/etpan.a(lib___etpan.static.o): in function `imap_full1':
etpan.c:(.text+0x3b3): undefined reference to `mailimap_new'
/usr/bin/ld: etpan.c:(.text+0x3c8): undefined reference to `mailimap_ssl_connect'
/usr/bin/ld: etpan.c:(.text+0x3fa): undefined reference to `mailimap_login'
/usr/bin/ld: etpan.c:(.text+0x410): undefined reference to `mailimap_select'
/usr/bin/ld: etpan.c:(.text+0x425): undefined reference to `mailimap_set_new_interval'
/usr/bin/ld: etpan.c:(.text+0x42d): undefined reference to `mailimap_fetch_type_new_fetch_att_list_empty'
/usr/bin/ld: etpan.c:(.text+0x435): undefined reference to `mailimap_fetch_att_new_uid'
/usr/bin/ld: etpan.c:(.text+0x440): undefined reference to `mailimap_fetch_type_new_fetch_att_list_add'
/usr/bin/ld: etpan.c:(.text+0x453): undefined reference to `mailimap_fetch'
/usr/bin/ld: etpan.c:(.text+0x4c6): undefined reference to `mailimap_fetch_list_free'
/usr/bin/ld: etpan.c:(.text+0x4ce): undefined reference to `mailimap_logout'
/usr/bin/ld: etpan.c:(.text+0x4d6): undefined reference to `mailimap_free'
collect2: error: ld returned 1 exit status
error: command exited with status 1
```
Note that the underlying `libetpan` wrapper itself built fine (using the `declare-native` statement).  It is the final linking of the executable with the wrapper which fails to link (due to missing cflags `-letpan` specification not getting carried through within `declare-executable` ).

A look at the JPM source code shows the problem in the `declare-executable` method which does not require `:cflags` parameter and hence does not pass `cflags` to the call to `create-executable` - 
```clojure
(defn declare-executable
    ...
    ...
  [&keys {:install install :name name :entry entry :headers headers}]
    ...
    ...
  (create-executable @{} entry dest)
    ...
    ...
)
```

An immediate fix for the problem is to include `:cflags` in the `declare-executable` arguments and also pass it along to the `create-executable` call  as follows - 
```clojure
(defn declare-executable
    ...
    ...
  [&keys {:install install :name name :entry entry :headers headers :cflags cflags}]
    ...
    ...
  (create-executable @{:cflags cflags} entry dest)
    ...
    ...
)
```

The above change in the JPM code worked for me, and the build for the native executable succeeds. 

 I am new to Janet so I  may not be the best person to recommend permanent Janet/JPM changes.  Please evaluate the above fix and update upstream code as necessary.

Also please evaluate if an equivalent linker specific parameter, e.g. `:lflags` could also be passed along for both `declare-executable` and `declare-native` methods, without cobblering the existing linker options required for successful compilation.
Thank you.

**Comment by bakpakin** [comment]
Thanks for the suggestions, this is definitely something that can be added. I see no reason to not allow passing in custom cflags and lflags here. `declare-native` should already have correct flag handling here.

To not clobber existing linker and C flags, jpm exposes default flags here. For example (with the proposed change), 

```
(declare-executable
  :name "beep"
  :cflags [;default-cflags "-fsanitize=address"]
  :lflags [;default-lflags "-lcurl"])
```

**Comment by bakpakin** [comment]
This should be fixed as of b8df47e063c7eaeb4b4de677141dca5948fa851a. You can copy the `jpm` file from the latest master (https://raw.githubusercontent.com/janet-lang/janet/master/auxbin/jpm) to your local jpm install to get these changes if you don't want to run from master.

**Comment by kixus** [comment]
@bakpakin  This is awesome!  Thank you for the prompt response.
Much appreciated.
I will try to look at the new code over the weekend.  In the meantime if you believe it is fixed, feel free to close the ticket.
Thanks again.
-kixus.
